File: Day_28/Solution.js
class Solution {

    maxDistance(arr) {
        // Use of Map here first
        let map = new Map();

        // We need to find the length of arr and initialize ans to 0
        let n = arr.length, ans = 0;

        // We need a for loop now
        for (let i = 0; i < n; i++) {

            // If the element is present or map contains the element
            if (map.has(arr[i])) {

                // Current index - previous index present in map
                let diff = i - map.get(arr[i]);

                // Update the answer
                ans = Math.max(ans, diff);
            } else {

                // If not present, then add current element with index
                map.set(arr[i], i);
            }
        }

        return ans;
    }
}

File: Day_28/Solution.java
class Solution {
    public int maxDistance(int[] arr) {
        Map<Integer,Integer> map  = new HashMap<>();
        int n  =arr.length, ans=0;
        for(int i=0;i<n;i++){
            if(map.containsKey(arr[i])){
                int diff = i-map.get(arr[i]);
                ans=  Math.max(ans,diff);
            }else{
                map.put(arr[i],i);
            }
        }
        return ans;
    }
}